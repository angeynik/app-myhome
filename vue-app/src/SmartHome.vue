<template>
    <div class="header-bottom">
      <nav>
              <router-link to="/login">Login</router-link> |
              <router-link to="/dashboard">Dashboard</router-link> |
              <router-link to="/smart-home">Smart Home</router-link> |
              <router-link to="/manufact-automatation">Automation</router-link> |
                  <!-- Profile (доступна для level 2 и 3) -->
              <router-link v-if="userLevel >= 2" to="/profile">Profile</router-link>
                  <!-- Users (доступна для level 3) -->
              <router-link v-if="userLevel >= 3" to="/users">Users</router-link>
            </nav>
      </div>
      <p style="width: 100%; height: 1px; background-color: var(--orange);"> </p>
      <div>
      <h1>Smart Home</h1>
      <p>Welcome to the Smart Home page!</p>
    </div>
  </template>

  <script>
  import { mapGetters } from 'vuex'; 
  export default {
    name: 'SmartHome',
    computed: {
    ...mapGetters(['level']), // Получаем уровень доступа из Vuex
    userLevel() {
      console.log('Уровень доступа пользователя:', this.level);
      return this.level || 0; // Если уровень не задан, считаем его равным 0
    },
  },
  };
  </script>
  
  <style scoped>
  /* Стили для SmartHome */
  </style>