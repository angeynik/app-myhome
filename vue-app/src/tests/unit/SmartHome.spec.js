import { mount } from '@vue/test-utils';
import SmartHome from '@/SmartHome.vue';

describe('SmartHome.vue', () => {
  it('matches the snapshot', () => {
    const wrapper = mount(SmartHome);
    expect(wrapper.html()).toMatchSnapshot();
  });
});