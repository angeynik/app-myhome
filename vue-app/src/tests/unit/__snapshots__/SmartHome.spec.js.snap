// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmartHome.vue matches the snapshot 1`] = `
"<div>
  <div class="header-bottom">
    <nav>
      <router-link-stub to="/login"></router-link-stub> | <router-link-stub to="/dashboard"></router-link-stub> | <router-link-stub to="/smart-home"></router-link-stub> | <router-link-stub to="/manufact-automatation"></router-link-stub> |
      <!-- Profile (доступна для level 2 и 3) -->
      <!--v-if-->
      <!-- Users (доступна для level 3) -->
      <!--v-if-->
    </nav>
  </div>
  <p style="width: 100%; height: 1px;"></p>
  <div>
    <h1>Smart Home</h1>
    <p>Welcome to the Smart Home page!</p>
  </div>
</div>"
`;
